main/ahb_for_sles.go

```diff
@@ -0,0 +1,249 @@
+  PublicCloudService             string
+  RegisterCloudGuestPath         string
+  RegionSrvMinVer                string
+  RegionSrvEnablerService        string
+  RegionSrv                      string
+  RegionSrvAddOn                 string
+  AddonPath                      string
+  RepoAlias                      string
+  ModName                        string
+  RepoUrl                        string
+}
+
+func parseCfg(filename string) (map[string]map[string]string, error) {
+    ini := make(map[string]map[string]string)
+    var head string
+
+    fh, err := os.Open(filename)
+    if err != nil {
+        return ini, fmt.Errorf("Could not open file '%v': %v", filename, err)
+    }
+    sectionHead := regexp.MustCompile(`^\[([^]]*)\]\s*$`)
+    keyValue := regexp.MustCompile(`^(\w*)\s*=\s*(.*?)\s*$`)
+    reader := bufio.NewReader(fh)
+    for {
+        line, _ := reader.ReadString('\n')
+        result := sectionHead.FindStringSubmatch(line)
+        if len(result) > 0 {
+            head = result[1]
+            ini[head] = make(map[string]string)
+            continue
+        }
+
+        result = keyValue.FindStringSubmatch(line)
+        if len(result) > 0 {
+            key, value := result[1], result[2]
+            ini[head][key] = value
+            continue
+        }
+
+        if line == "" {
+            break
+        }
+    }
+
+    return ini, nil
+}
+
+func _isRegistered() (bool, error) {
+  services, err := filepath.Glob("/etc/zypp/services.d/*.service")
+  if err != nil {
+    fmt.Println(err)
+    return false, err
+  }
+  return len(services) > 0, err
+}
+
+func _hasPubCloudMod(pubCloudService string) (bool, string, error) {
+  services, err := filepath.Glob("/etc/zypp/services.d/*.service")
+  if err != nil {
+    return false, "", err
+  }
+  hasPubCloud := false
+  version := ""
+
+  for _, service := range services {
+    serviceFile, _ := parseCfg(service)
+    for _, value := range serviceFile {
+      if strings.Contains(strings.ToLower(value["url"]), pubCloudService) {
+        hasPubCloud = true
+      } else {
+        if version == "" {
+          if strings.Contains(value["repo_1"], "SLE-15") {
+            version = "15"
+          }
+          if strings.Contains(value["repo_1"], "SLE-12") {
+            version = "12"
+          }
+        }
+      }
+    }
+  }
+  return hasPubCloud, version, nil
+}
+
+func getAhbInfo() (AHBInfo) {
+  return AHBInfo{
+    PublicCloudService: "public_cloud",
+    RegisterCloudGuestPath: "/usr/sbin/registercloudguest",
+    RegionSrvMinVer: "9.3.1",
+    RegionSrvEnablerService: "regionsrv-enabler-azure.service",
+    RegionSrv: "cloud-regionsrv-client",
+    RegionSrvAddOn: "cloud-regionsrv-client-addon-azure",
+    AddonPath: "/usrb/sbin/regionsrv-enabler-azure",
+    RepoAlias: "sle-ahb-packages",
+    ModName: "sle-module-public-cloud",
+    RepoUrl: "https://updates.suse.com/SUSE/Updates/SLE-Module-Public-Cloud-Unrestricted/%s/x86_64/update",
+  }
```

Comment:
We need to track the additional packages that may need installation if they are not there:

cloud-regionsrv-client-plugin-azure
regionServiceClientConfigAzure
regionServiceCertsAzure

When a customer brings their own image and wants to turn it into a PAYG instance it is unlikely they will have those packages.

Reply:
Done